### PR TITLE
Implement Superchain icon modal + tooltip interaction

### PR DESCRIPTION
**Pull request by imranjami**
### Description
Implemented the superchain icon modal + tooltip interaction on the dapp console homepage. Both are using our `ui-components` from the design system.

Also brought in the dates + urls for docs items from the [PRD doc here](https://docs.google.com/document/d/1iJeKa9o3cRnIO0yxxwnDgFaP8nr_hocsBKr6tDlWKD0/edit)

https://github.com/ethereum-optimism/ecosystem/assets/10838132/be995da0-1407-454a-9d01-c18f5d251cdc

### CODE REVIEW

**Comment by imranjami** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/119?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #107** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/107?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #108** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/108?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #113** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/113?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #119** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/119?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/119?utm_source=stack-comment).

File: apps/dapp-console/app/console/components/ProjectIconLinks.tsx [hunk 1]
```diff
@@ -3,39 +3,62 @@
...
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogTrigger,
+} from '@eth-optimism/ui-components/src/components/ui/dialog'
+import { Button } from '@eth-optimism/ui-components/src/components/ui/button'
+import { Text } from '@eth-optimism/ui-components/src/components/ui/text'
 
 const linkItems = [
```
Comment (tarunkhasnavis) [inline apps/dapp-console/app/console/components/ProjectIconLinks.tsx:22]: Would we want to put this in a constants file?

File: apps/dapp-console/app/constants/index.tsx [hunk 2]
```diff
@@ -36,35 +36,35 @@ const externalRoutes: Routes = {
   },
   // Documentation items
   ETH_DOCS: {
-    path: '',
+    path: 'https://ethereum.org/developers/docs',
```
Comment (tarunkhasnavis) [inline apps/dapp-console/app/constants/index.tsx:39]: what happens when this constants file gets bloated?

**Review by tarunkhasnavis — APPROVED** [review]

File: apps/dapp-console/app/console/components/ProjectIconLinks.tsx (hunk 1 above)
Comment (imranjami) [reply apps/dapp-console/app/console/components/ProjectIconLinks.tsx:22]: moved it there in a later PR

File: apps/dapp-console/app/constants/index.tsx (hunk 2 above)
Comment (imranjami) [reply apps/dapp-console/app/constants/index.tsx:39]: We can separate things out later on if needed. For now there isn't enough to split things up

**Pull request merged**